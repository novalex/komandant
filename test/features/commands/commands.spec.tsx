import React from 'react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { BrowserRouter as Router } from 'react-router-dom';

import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Commands from '../../../app/features/commands';
import reducer, { initialState } from '../../../app/features/commands/slice';

Enzyme.configure({ adapter: new Adapter() });

function setup(
	preloadedState: { commands: typeof initialState } = {
		commands: initialState,
	}
) {
	const store = configureStore({
		reducer: { commands: reducer },
		preloadedState,
	});

	const getWrapper = () =>
		mount(
			<Provider store={store}>
				<Router>
					<Commands />
				</Router>
			</Provider>
		);
	const component = getWrapper();
	return {
		store,
		component,
	};
}

describe('component commands', () => {
	it('should match exact snapshot', () => {
		const { store } = setup();
		const tree = renderer
			.create(
				<Provider store={store}>
					<Router>
						<Commands />
					</Router>
				</Provider>
			)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
