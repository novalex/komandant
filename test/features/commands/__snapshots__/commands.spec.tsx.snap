// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component commands should match exact snapshot 1`] = `
<div
  className="commandsWrap"
  data-tid="commands"
>
  <div
    className="commandsEmpty"
  >
    <p>
      No commands yet!
    </p>
    <a
      href="/new-command"
      onClick={[Function]}
    >
      Add new command
    </a>
  </div>
</div>
`;
